### PR TITLE
Tests: Unit Tests for Math

### PR DESCRIPTION
**Pull request by epreston**
Related issue: #XXXX

**Description**

This cleans up the unit tests for Math.
Adds missing unit test files, fills in some unit tests, populates the missing member tests with stubs.

A test fixture, 'math/Constants.tests.js' wasnt a test file but a set of fixures used by:

animation/AnimationMixer.tests.js
core/BufferGeometry.tests.js
core/Object3D.tests.js
core/Uniform.tests.js
math/Box2.tests.js
math/Box3.tests.js
math/Color.tests.js
math/Cylindrical.tests.js
math/Euler.tests.js
math/Frustum.tests.js
math/Line3.tests.js
math/Matrix4.tests.js
math/Plane.tests.js
math/Quaternion.tests.js
math/Ray.tests.js
math/Sphere.tests.js
math/Spherical.tests.js
math/Triangle.tests.js
math/Vector2.tests.js
math/Vector3.tests.js
math/Vector4.tests.js

Shared math constants for tests moved to utils.  Only tests in test source.  This also removes the confusion with the actual 'Constants.tests.js' file in the parent directory.

### CODE REVIEW

File: test/unit/src/math/Interpolant.tests.js [hunk 1]
```diff
@@ -54,20 +54,52 @@
 		// Tests
 
 		// INSTANCING
-		QUnit.todo( 'Instancing', ( assert ) => {
+		QUnit.test( 'Instancing', ( assert ) => {
+
+			var interpolant = new Mock( null, [ 1, 11, 2, 22, 3, 33 ], 2, [] );
+			assert.ok( interpolant !== undefined, 'instanced.' );
```
[the remark this answers is not in the document]
Comment (epreston) [reply test/unit/src/math/Interpolant.tests.js]: I agree.  Better test would be:
```js
assert.strictEqual(
	interpolant instanceof Interpolant, true,
	'Mock extends from Interpolant'
);
```

**Comment by epreston** [comment]
Are we running 8 of the same E2E tests ?  Each operating system appears 8 times.

**Comment by LeviPesin** [comment]
Yes, we run 8 tests on each of 3 OSes.

**Comment by epreston** [comment]
What's the difference between them ?  Is it to find variance in images generated ?

**Comment by LeviPesin** [comment]
Each runs on the 1/8'th of all examples -- it would be too slow to combine them into one test without parallelization.

**Pull request merged**